### PR TITLE
Support binding to network interfaces on Windows

### LINKED ISSUE #871: Bind Nicotine+ to specified network adapter
I'd like to be able to choose a network adapter to bind Nicotine+ to.

This is useful for VPNs, etc. qBittorrent has implemented this very nicely as an advanced preference via a simple dropdown menu.

### PR DESCRIPTION
**Pull request by mathiascode**
Closes #871

### CODE REVIEW

File: pynicotine/slskproto.py [hunk 1]
```diff
@@ -311,44 +317,159 @@ def _bind_listen_port(self):
...
 
-            ip_address = socket.inet_ntoa(ip_if[20:24])
+        for _i, interface_name in interface_name_index:
+            try:
+                import fcntl  # pylint: disable=import-error
 
-        except ImportError:
-            ip_address = None
+                with socket.socket(socket.AF_INET, socket.SOCK_DGRAM) as sock:
+                    siocgifaddr = 0x8915 if sys.platform == "linux" else 0xc0206921  # 0xc0206921 for *BSD, macOS
```
Comment (slook) [inline pynicotine/slskproto.py:439]: Why is not constant anymore?

Also, fnctl only needs to be imported once.

Comment (mathiascode) [reply pynicotine/slskproto.py:439]: > Also, fnctl only needs to be imported once.

I can't import fnctl globally, it's not available on Windows.

**Comment by mathiascode** [comment]
@slook Windows looks good, can you check that everything works fine in Linux? I'm still waiting for parts to repair my main computer, so I can't do it myself right now.

**Comment by slook** [comment]
I've not tried anything complex, but it does appear to work as intended.

In the preferences dialog if an interface is configured the incorrect list item is highlighted upon opening the dropdown for the first time, but that minor accessibility problem is probably unrelated to the changes in this PR.

**Pull request merged**